components/config/prefs.rs

```diff
@@ -16,8 +16,8 @@ use crate::pref_util::Preferences;
 pub use crate::pref_util::{PrefError, PrefValue};
 
 static PREFS: LazyLock<Preferences<'static, Prefs>> = LazyLock::new(|| {
-    let def_prefs: Prefs = serde_json::from_str(&resources::read_string(Resource::Preferences))
-        .expect("Failed to initialize config preferences.");
+    let def_prefs: Prefs =
+        serde_json::from_str(&resources::read_string(Resource::Preferences)).unwrap_or_default();
```

Comment:
Should we log the error somewhere so it doesn't cause confusion when the preference doesn't match what is expected?

Reply:
I feel like we should also set sensible defaults to `Preferences`. How about adding a warning log here and opening a follow up issue to set sensible defaults?

Reply:
I think this should be an 'error' log since the resource is present, but the failure can also be due to incorrect syntax. My preference would be to make the 'no prefs.json resource, so use defaults' case explicit in the embedder API without using an empty string to represent that. I don't have a good answer for what that API should look like though. Perhaps @mrobinson has some ideas.

Reply:
Let's open an issue to discuss then. Whether to keep the json and whether to set the default in `servo_config`.
I'll add an error log here instead.